### PR TITLE
Correctly Assign Variables to Tables for Similar Names

### LINKED ISSUE #10: [BUG] REDCap Form Names can be "contained" in other names
# Expected Behavior
Varying REDCap variables should exist solely in their expected instrument tables when output by `read_redcap_tidy`.

# Current Behavior
There is a minor bug that appeared when evaluating the outputs of the "repeat/nonrepeat" REDCap test database. This results in `nonrepeat*` columns appearing in the `repeated` table output because `repeat` as a word is "contained" in `nonrepeat`.

## How to Reproduce the Bug:
1) Using the current `REDCapTidieR` functions, use the repeat/nonrepeat REDCap test database and observe the existence of `nonrepeat*` columns in the `repeated` table output.

**Screenshots**
```r
> read_redcap_tidy(redcap_uri = Sys.getenv("REDCAP_URI"), 
+                  token = Sys.getenv("REDCAPTIDIER_LONGITUDINAL_API")) %>% bind_tables()
> repeated
# A tibble: 9 × 10
  record_id redcap_repeat_instance redcap_event redcap_arm nonrepeat_1 repeat_1 nonrepeat_2 repeat_2 nonrepeat…¹ form_…²
      <dbl>                  <dbl> <chr>             <int> <chr>       <chr>    <chr>       <chr>          <dbl>   <dbl>
1         1                      1 event_1               1 NA          1        NA          2                 NA       0
2         1                      2 event_1               1 NA          3        NA          4                 NA       0
3         1                      3 event_1               1 NA          5        NA          6                 NA       0
4         1                      1 event_2               1 NA          A        NA          B                 NA       0
5         1                      2 event_2               1 NA          C        NA          D                 NA       0
6         3                      1 event_1               1 NA          C        NA          D                 NA       0
7         3                      1 event_2               1 NA          E        NA          F                 NA       0
8         3                      2 event_2               1 NA          G        NA          H                 NA       0
9         4                      1 event_3               2 NA          R1       NA          R2                NA       0
# … with abbreviated variable names ¹​nonrepeated_complete, ²​form_status_complete
```

## Proposed Solution

Replacing instances of `contains` with `starts_with` in the `extract_*` functions resolves this problem.

### Checklist
Before submitting this issue, please check and verify below that the submission meets the below criteria:

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by rsh52**
# Description
Use of the `contains` function in `extract_*_long` functions resulted in `repeat` variable names appear in `nonrepeat` tables since the former word exists in the latter.

See Issue #10 for more information.

# Proposed Changes 
List changes below in bullet format:

- Replace 2 instances of `contains` with `starts_with` to correct behavior

**Screenshots**
```r
> read_redcap_tidy(redcap_uri = Sys.getenv("REDCAP_URI"), 
+                  token = Sys.getenv("REDCAPTIDIER_LONGITUDINAL_API")) %>% bind_tables()

> repeated
# A tibble: 9 × 7
  record_id redcap_repeat_instance redcap_event redcap_arm repeat_1 repeat_2 form_status_complete
      <dbl>                  <dbl> <chr>             <int> <chr>    <chr>                   <dbl>
1         1                      1 event_1               1 1        2                           0
2         1                      2 event_1               1 3        4                           0
3         1                      3 event_1               1 5        6                           0
4         1                      1 event_2               1 A        B                           0
5         1                      2 event_2               1 C        D                           0
6         3                      1 event_1               1 C        D                           0
7         3                      1 event_2               1 E        F                           0
8         3                      2 event_2               1 G        H                           0
9         4                      1 event_3               2 R1       R2                          0
```

### Issue Addressed
**Closes #10**

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [ ] New/revised functions have associated tests
   > No test currently made, let me know if you think it would necessary to test for like names in variables.
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

**Comment by skadauke** [comment]
This will still break if you have an field named `repeat_22` in another instrument. I think we need to handle discovering columns derived from multi-checkboxes explicitly.

**Comment by rsh52** [comment]
Drats, alright sounds good!

**Comment by rsh52** [comment]
I haven't fully thought this through yet, but maybe we could develop a way of initially encoding column names early in the call (at simplest `1`:`n`) so they're guaranteed to be unique and then use `match` or similar by the time the `extract_*` functions do their magic.

**Comment by rsh52** [comment]
@skadauke, the latest commit passes all checks and introduces two new helper/utli functions:

- `checkbox_appender`: responsible for enforcing regex via `gsub` to take metadata associated with checkboxes and return a vector of appended checkbox variable names
- `update_field_names`: applied at the start of `read_redcap_tidy` and responsible for applying `checkbox_appender` to the metadata object and `unnest`ing the returned values to make individual checkbox `field_name` rows in the `metadata` object

Regex is not a strong suit for me and I would like to make these functions more elegant but for now they work the way I expect/want them to. 

Thoughts appreciated when you're feeling better!

**Review by skadauke — COMMENTED** [review]
Hi - this looks good! Do we need a test to make sure that the expected checkbox columns show? Also, please take a look at my comment for a proposed `parse_label` helper function. If you implement this from the second code block I've written in the comment, I think you should be able to refactor the code in `checkbox_appender` to use it pretty quickly, and also be in a good spot to tackle issue #8 in a future PR.

File: R/utils.R [hunk 1]
```diff
@@ -62,3 +62,61 @@ link_arms <- function(
...
+#' @param field_name The \code{db_metadata$field_name} to append onto the string
+#' @param string A \code{db_metadata$select_choices_or_calculations} field pre-filtered for checkbox \code{field_type}
+#'
+#' @import dplyr
+#' @keywords internal
+
+checkbox_appender <- function(field_name, string){
+  prefix <- paste0(field_name, "___")
+
+  out <- string %>%
```
[the remark this answers is not in the document]
Comment (rsh52) [reply R/utils.R:79]: Awesome! I like this much more, I wanted to get away from using much regex if possible but I've never really used `matrix` in R. This is a great use case for it.

New commit incoming, let me know what you think. Tried putting some room in for `raw_or_label` arguments for the future.

**Review by skadauke — APPROVED** [review]
Looks good to me! 👍
